Cargo.toml

```diff
@@ -13,6 +13,7 @@ unused_lifetimes                       = "warn"
 unused_macro_rules                     = "warn"
 
 [workspace.lints.clippy]
+allow_attributes        = "deny"
```

Comment:
I'm not entirely sure if we should outright block allow. We might want to allow `allow` in generated code to minimize new surprise diagnostics from clippy when we upgrade rust versions. Could we perhaps do that on a module level for generated code and block it everywhere else?

Reply:
@dyc3 - I don't know-- my changes are finally finalized with `expect` globally disabled. The diff is +288, -2400, mostly removing unneeded `allow`s. My intuition says that most of the pain of switching to `expect` is contained in this repo, and that `expect` will work fine for future Clippy checks we wish to disable.

Reply:
And it's ready for review now, @dyc3! Thank you for taking a look.

Reply:
@dyc3 I left this here. I think it works out OK because this lint does not prevent `#![allow(...)]` in individual code files. Let me know if you disagree though. I reverted many of my tweaks to the code generation process, although the process did help me implement a few lints on generated code still.